### ISSUE TITLE
Define a governance structure

### ISSUE THREAD

**Issue by djhoese** [vispy/vispy#2128]
We should consider creating a governance description for the VisPy project. A document like this can help newcomers to the project understand how decisions are made and to help define how disagreements should be resolved when they pop up. There were a couple PRs in this last 0.7 release that resulted in some differing opinions and overall we were able to come to an agreement, but in some cases it was more of a consensus between 2 or more developers and the others were left to "just deal with it". I'm sure there have been cases where I, as someone who has the time and funding to participate in this project, merged a PR or closed or commented on an issue that other people who care about this project didn't agree with. Part of this is likely due to me rushing parts of VisPy's development and release schedule when I feel like it (bad habit).

This issue is to discuss whether the core contributors of vispy have an issue with defining a document like this. In the document we could define various roles of community members (core devs versus regular contributors versus issue reviewers, etc). We could also define how disagreements are resolved and maybe to fix my bad habits a minimum "feature freeze" period and minimum "feedback requested" period.

Here is scikit-image's governance document as an example (@jni any others that would be good for inspiration): https://scikit-image.org/docs/dev/skips/1-governance.html

I think the thing to keep in mind when talking about this is that VisPy currently has low maintainer participation. That may change as datoviz/py-wgpu/vulkan integrations get going, but those steps will also probably result in more disagreements about the future of VisPy. We don't have to make our governance structure match any other project's structure. We have a special (few contributors/maintainers) situation here and can make the governance fit however we need it to. The point is that we define how things work so we can refer to it when questions get raised.

Thoughts?

**Comment by rossant** [comment]
Agree that this is important. I like the scikit-image model. Perhaps we could start from there. One avenue of simplification for VisPy might be to merge the core developers with the steering council?

**Comment by almarklein** [comment]
I agree that formalizing this may help make things more clear, and in some cases more fair. I also think that the scikit-image model would be a good starting point. I would still separate the steering council from core devs though, otherwise I fear we may be more reluctant to include more core devs (and we should have more). As I understand it, the steering council only matters in decisions that affect the high-level direction of vispy.

**Comment by djhoese** [comment]
There is a new "Minimal Working Governance" project that seeks to define a starting point for governance structure: https://github.blog/2021-07-22-minimum-viable-governance-lightweight-community-structure-foss-projects/

I haven't read the actual document, only this blog post, but we could probably adopt this as a starting point.

**Comment by rougier** [comment]
I agree it might be a good start. Here is the document https://github.com/github/MVG/blob/main/org-docs/CHARTER.md

**Comment by djhoese** [comment]
See #2367 for first draft.
